massa-bootstrap/src/server.rs

```diff
@@ -187,14 +188,38 @@ impl BootstrapServer {
                 }
 
                 // listener
-                res_connection = listener.accept(&whitelist, &blacklist) => {
-                    let (dplx, remote_addr) = if res_connection.is_ok() {
-                        res_connection.unwrap()
+                res_connection = listener.accept() => {
+                        let (mut server, remote_addr) = if res_connection.is_ok() {
+                            let (dplx, remote_addr) = res_connection.unwrap();
+                            let server = BootstrapServerBinder::new(
+                                dplx,
+                                self.keypair.clone(),
+                                self.bootstrap_config.max_bytes_read_write,
+                                self.bootstrap_config.max_bootstrap_message_size,
+                                self.bootstrap_config.thread_count,
+                                self.bootstrap_config.max_datastore_key_length,
+                                self.bootstrap_config.randomness_size_bytes,
+                                self.bootstrap_config.consensus_bootstrap_part_size);
+                            // we didn't test whether Ip is allowed
+                        #[cfg(test)]
+                        {
+                            (server, remote_addr)
+                        }
+
+                        #[cfg(not(test))]
+                        {
+                            // check whether incoming peer IP is allowed or return an error which is ignored
+                            let res = self.is_ip_allowed(remote_addr, server, &whitelist, &blacklist).await;
+                            if res.is_ok() {
+                                res.unwrap()
+                            } else {
+                                continue
+                            }
```

Comment:
This might be a nice place to use let-else:

```suggestion
                            let Ok(res) = self.is_ip_allowed(remote_addr, server, &whitelist, &blacklist).await else {
                                continue;
                            };
```

Reply:
(wrote this out manually in github. Not sure if the `;` on line 213 is needed)

Reply:
```
                        let (mut server, remote_addr) = if let Ok(res) = res_connection {
                            let server = BootstrapServerBinder::new(
                                res.0,
                                self.keypair.clone(),
                                self.bootstrap_config.max_bytes_read_write,
                                self.bootstrap_config.max_bootstrap_message_size,
                                self.bootstrap_config.thread_count,
                                self.bootstrap_config.max_datastore_key_length,
                                self.bootstrap_config.randomness_size_bytes,
                                self.bootstrap_config.consensus_bootstrap_part_size);
                        // we didn't test whether Ip is allowed
                        #[cfg(test)]
                        {
                            (server, res.1)
                        }

                        #[cfg(not(test))]
                        {
                            // check whether incoming peer IP is allowed or return an error which is ignored
                            let Ok(res) = self.is_ip_allowed(res.1, server, &whitelist, &blacklist).await else {
                                continue;
                            };
                            res
                        }
                    } else {
                        continue
                    };
```

@Eitu33 @Ben-PH is it more readable like this ? (not sure)

Reply:
done

Reply:
I think so, yeah. The dropping of the `unwrap` is nice in the old line 193 is a nice extra.

Reply:
IMHO, it's already done ✅